ADBDEV-4726-34-2 Fix null-check for elems array

dnskvlnk opened the pull request:
Fix null-check for elems array

construct_md_array passed elems to CopyArrayEls where it aws dereferenced
without null-check. This patch changes the condition to prevent it.

Commit:
Test without fixedwidthtype

RekGRpth commented:
If you wanted to run all the tests, then you had to put the base branch in adb-6.x-dev.

Commit:
Fix null-check for elems array

construct_md_array passed elems to CopyArrayEls where it aws dereferenced
without null-check. This patch changes the condition to prevent it.

Commit:
Rerun tests

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/64909

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1083323

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1083317

dnskvlnk commented:
@RekGRpth tests fail with that approach, I think it's better to use Insist to not break anything

BenderArenadata commented:
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1083315

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1083324

BenderArenadata commented:
Failed job Regression tests with Postgres on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1083316

RekGRpth commented:
> @RekGRpth tests fail with that approach, I think it's better to use Insist to not break anything

but with
```diff
diff --git a/src/backend/utils/adt/arrayfuncs.c b/src/backend/utils/adt/arrayfuncs.c
index e5e5d0c8a27..faa6cddcd81 100644
--- a/src/backend/utils/adt/arrayfuncs.c
+++ b/src/backend/utils/adt/arrayfuncs.c
@@ -3064,7 +3064,7 @@ construct_md_array(Datum *elems,
 	{
 		/* do nothing */
 	} 
-	else
+	else if (elems!=NULL)
 	{
 		CopyArrayEls(result,
 					 elems, nulls, nelems,
```
triggers test passed

BenderArenadata commented:
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1083318

RekGRpth commented:
try test only one change to branch adb-6.x-dev

dnskvlnk commented:
@RekGRpth is there anything that needs attention in this pr?

Commit:
Remove extra space
